### PR TITLE
Add tests for invalid requests sent from the client to the http listneer

### PR DESCRIPTION
**Pull request by hughbe**
Contributes to #13618

### CODE REVIEW

File: src/System.Net.HttpListener/tests/HttpListenerRequestTests.cs [hunk 1]
```diff
@@ -90,18 +90,27 @@ public static IEnumerable<object[]> ContentEncoding_TestData()
...
                 Assert.Equal(Encoding.Default, request.ContentEncoding);
-            }, sendContent: false);
+            }, content: null);
         }
 
-        [ConditionalTheory(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotOneCoreUAP))]
-        [InlineData("Content-Length: 0", 0)]
-        [InlineData("Transfer-Encoding: chunked", -1)]
-        public async Task ContentLength_GetProperty_ReturnsExpected(string contentLengthString, long expected)
+        [Theory]//[ConditionalTheory(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotOneCoreUAP))]
```
Comment (stephentoub) [inline src/System.Net.HttpListener/tests/HttpListenerRequestTests.cs]: What is this comment for?

File: src/System.Net.HttpListener/tests/HttpListenerRequestTests.cs [hunk 2]
```diff
@@ -171,8 +180,7 @@ public static IEnumerable<object[]> ContentEncoding_TestData()
             });
         }
         
-        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotOneCoreUAP))]
-        [PlatformSpecific(TestPlatforms.Windows)]
+        [ConditionalFact(nameof(Helpers) + "." + nameof(Helpers.IsWindowsImplementation))]
```
Comment (stephentoub) [inline src/System.Net.HttpListener/tests/HttpListenerRequestTests.cs]: Is there an issue number for this?  Let's try to ensure that anywhere we're special-casing out a platform, we have a comment that indicates why, either an issue number if it's temporary or a short-description of why it's permanently platform-specific.

File: src/System.Net.HttpListener/tests/HttpListenerRequestTests.cs [hunk 3]
```diff
@@ -242,8 +250,8 @@ public static IEnumerable<object[]> ContentEncoding_TestData()
             });
         }
 
-        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotOneCoreUAP))]
-        [PlatformSpecific(TestPlatforms.Windows)] // We get the ClientCertificate during connection on Unix.
+        // We get the ClientCertificate during connection on Unix.
```
Comment (stephentoub) [inline src/System.Net.HttpListener/tests/HttpListenerRequestTests.cs]: Issue number?

File: src/System.Net.HttpListener/tests/HttpListenerRequestTests.cs [hunk 4]
```diff
@@ -320,8 +328,7 @@ public static IEnumerable<object[]> ContentEncoding_TestData()
             });
         }
 
-        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotOneCoreUAP))]
-        [PlatformSpecific(TestPlatforms.Windows)]
+        [ConditionalFact(nameof(Helpers) + "." + nameof(Helpers.IsWindowsImplementation))]
```
Comment (stephentoub) [inline src/System.Net.HttpListener/tests/HttpListenerRequestTests.cs]: Issue number? Or a comment about why it's platform-specific?  I'll stop commenting on these.

File: src/System.Net.HttpListener/tests/InvalidClientRequestTests.cs [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+                    host = host ?? $"Host: {listeningUri.Host}\r\n";
+                    string fullRequest = $"{requestLineWithPathAndQuery}\r\n{host}{string.Join("\r\n", headers ?? new string[0])}{content}\r\n";
+                    
+                    Task<HttpListenerContext> serverTask = Factory.GetListener().GetContextAsync();
+                    client.Send(Encoding.Default.GetBytes(fullRequest));
+
+                    byte[] errorMessageBytes = new byte[512];
+                    Task<int> clientTask = Task.Run(() => client.Receive(errorMessageBytes));
+
+                    Task completedTask = await Task.WhenAny(clientTask, serverTask);
```
Comment (stephentoub) [inline src/System.Net.HttpListener/tests/InvalidClientRequestTests.cs]: It'd probably be good to add an assert verifying that serverTask doesn't complete.

Comment (hughbe) [inline src/System.Net.HttpListener/tests/InvalidClientRequestTests.cs]: Good idea - but how do I do this? Thanks

Comment (stephentoub) [inline src/System.Net.HttpListener/tests/InvalidClientRequestTests.cs]: You can do something like:
```C#
Assert.False(serverTask.IsCompleted, $"Server task was completed: {serverTask.Status}");
```

Comment (hughbe) [inline src/System.Net.HttpListener/tests/InvalidClientRequestTests.cs]: Oh right its that simple!

**Comment by stephentoub** [comment]
Thanks, @hughbe!

**Pull request merged**